Makefile.d/install-flannel.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -eu -o pipefail
+
+function INFO() {
+	echo >&2 -e "\e[104m\e[97m[INFO]\e[49m\e[39m $@"
+}
+function WARNING() {
+	echo >&2 -e "\e[101m\e[97m[WARNING]\e[49m\e[39m $@"
+}
+
+function ERROR() {
+	echo >&2 -e "\e[101m\e[97m[ERROR]\e[49m\e[39m $@"
+}
+
+# See chart values, 0 indicates default for platform
+# https://github.com/flannel-io/flannel/blob/v0.26.1/chart/kube-flannel/values.yaml
+: "${U7S_PORT_FLANNEL:='0'}"
+: "${U7S_PORT_ETCD:='2379'}"
+
+INFO "Flannel port: ${U7S_PORT_FLANNEL}"
+INFO "ETCD port: ${U7S_PORT_ETCD}"
+
+# Check hard dependency commands
+for cmd in helm kubectl git; do
+	if ! command -v "${cmd}" >/dev/null 2>&1; then
+		ERROR "Command \"${cmd}\" is not installed"
+		exit 1
+	fi
+done
+
+# We need to customize the values.yaml to expose the backendPort and args
+flannel_root=$(mktemp -d -u -t flannel-XXXXXXX)
+git clone --quiet --depth 1 --branch v0.26.1 https://github.com/flannel-io/flannel $flannel_root
+cd $flannel_root/chart
+
+# Write a new values.yaml that exposes what we need
+cat <<EOF > ./new-values.yaml
+---
+global:
+  imagePullSecrets:
+# - name: "a-secret-name"
+
+# The IPv4 cidr pool to create on startup if none exists. Pod IPs will be
+# chosen from this range.
+podCidr: "10.244.0.0/16"
+podCidrv6: ""
+flannel:
+  # kube-flannel image
+  image:
+    repository: docker.io/flannel/flannel
+    tag: v0.26.1
+  image_cni:
+    repository: docker.io/flannel/flannel-cni-plugin
+    tag: v1.5.1-flannel2
+  # flannel command arguments
+  enableNFTables: false
+  args:
+  - "--ip-masq"
+  - "--kube-subnet-mgr"
+  - "--etcd-endpoints=\"http://127.0.0.1:4001,http://127.0.0.1:${U7S_PORT_ETCD}\""
```

Comment:
This is unlikely to work when the node is not a control plane

Reply:
Are you sure? Wouldn't the lima tests fail if this were the case?

Reply:
The reason I did this is because it is the default - what is currently generated when we don't specify anything.

Reply:
I added `--etcd-endpoints=\"http://127.0.0.1:4001,https://${U7S_HOST_IP}:${U7S_PORT_ETCD},http://127.0.0.1:${U7S_PORT_ETCD}` to try, which keeps the defaults the version we have would generate, plus a variant that I think would be see-able if not running on the node.

Reply:
The etcd config should be just removed.

etcd isn't used here, as kube-subnet-mgr is enabled
https://github.com/flannel-io/flannel/blob/v0.26.1/Documentation/configuration.md

Reply:
okay will remove - but just to clarify, when we do that, it does default to the original value that I had set. Is that OK?

Reply:
okay I think I chose a good middle strategy - I commented it out, and left a note to the user that it isn't used with the link (your comment above). This way, if the developer user decides to edit the configuration and runs into an issue with etcd (having changed the port) they will have awareness into the issue. A lot of the problems I ran into were just needing to spend time figuring out command line flags (and how to provide them) so this might save someone time in the future!